tests/nosetests/pyanaconda_tests/module_subscription_tasks_tests.py

```diff
@@ -241,3 +246,149 @@ def set_rhsm_config_tast_restore_default_value_test(self):
+            # now mock the backend of the bus instance to prevent it from
+            # trying to get an actual connection
+            private_bus._provider = provider
+            # and try to get a proxy
+            private_register_proxy = private_bus.connection.get_proxy(RHSM.service_name,
+                                                                      RHSM_REGISTER.object_path)
+            # check correct address was used
+            get_address_bus.assert_called_with("abcdefgh")
+            # check correct proxy was requested from the connection
+            connection.get_proxy.assert_called_once_with(RHSM.service_name,
+                                                         RHSM_REGISTER.object_path)
+            self.assertEqual(private_register_proxy, connection.get_proxy.return_value)
+        # exit the context manager and check cleanup happened as expected
+        connection.disconnect.assert_called_once()
+        register_server_proxy.Stop.assert_called_once_with("en_US.UTF-8")
+
+
+class RegistrationTasksTestCase(unittest.TestCase):
+    """Test the registration tasks."""
+
+    @patch("os.environ.get", return_value="en_US.UTF-8")
+    @patch("pyanaconda.modules.subscription.runtime.RHSMPrivateBus")
+    def username_password_success_test(self, private_bus, environ_get):
+        """Test the RegisterWithUsernamePasswordTask - success."""
+        # register server proxy
+        register_server_proxy = Mock()
+        # private register proxy
+        get_proxy = private_bus.return_value.__enter__.return_value.get_proxy
+        private_register_proxy = get_proxy.return_value
+        # instantiate the task and run it
+        task = RegisterWithUsernamePasswordTask(rhsm_register_server_proxy=register_server_proxy,
+                                                username="foo_user",
+                                                password="bar_password")
+        self.assertTrue(task.run())
+        # check the private register proxy Register method was called correctly
+        private_register_proxy.Register.assert_called_once_with("",
+                                                                "foo_user",
+                                                                "bar_password",
+                                                                {},
+                                                                {},
+                                                                "en_US.UTF-8")
+
+    @patch("os.environ.get", return_value="en_US.UTF-8")
+    @patch("pyanaconda.modules.subscription.runtime.RHSMPrivateBus")
+    def username_password_failure_test(self, private_bus, environ_get):
+        """Test the RegisterWithUsernamePasswordTask - failure."""
+        # register server proxy
+        register_server_proxy = Mock()
+        # private register proxy
+        get_proxy = private_bus.return_value.__enter__.return_value.get_proxy
```

Comment:
I don't think this is how you are supposed to do it. But maybe I should stop giving you "ideas" ;-)